packages/babel-types/src/validators/isReferenced.js

```diff
@@ -134,6 +134,18 @@ export default function isReferenced(
     // no: type X = { NODE: OtherType }
     case "ObjectTypeProperty":
       return parent.key !== node;
+
+    // no: enum X { NODE }
+    case "TSEnumMember":
+      return false;
```

Comment:
I just noticed that we need to check that node is the `id`, because `enum X { foo = NODE }` is referenced.

Reply:
Good call, updated and added some extra tests.